### PR TITLE
Updated build instructions

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This applies for the build instructions for Zonemaster-LDNS. That repository has a submodule to LDNS. If the submodule is checked out and the version is changed in the branch checked out, then then submodule must be updated. If changing between branches with different version of the submodule, then the submodule must also be updated.

The changes in this PR assumes that https://github.com/zonemaster/zonemaster-ldns/pull/158 is merged.

## How to test this PR

Review and test using the documents.

### CODE REVIEW

**Comment by pnax** [comment]
I suggest a different approach based on `make` to automatically reconfigure the submodule, see https://github.com/zonemaster/zonemaster-ldns/pull/158.

**Comment by matsduf** [comment]
> I suggest a different approach based on `make` to automatically reconfigure the submodule, see [zonemaster/zonemaster-ldns#158](https://github.com/zonemaster/zonemaster-ldns/pull/158).

zonemaster/zonemaster-ldns#158 will not replace this PR, but if that is merged then it will be enough to run `git submodule update` instead of `git submodule deinit -f . ; git submodule update --init --recursive`

**Comment by matsduf** [comment]
@pnax, please review.

File: docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md [hunk 1]
```diff
@@ -36,14 +36,15 @@ the `develop` branch and that your clone is up-to-date.
        git fetch --all
        git branch
 
+For Zonemaster-LDNS only - reset submodule (LDNS):
```
Comment (pnax) [inline docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md:39]: For information, it's more an update of the submodule than a reset (the submodule is checked out to the right commit).

**Review by pnax — APPROVED** [review]

File: docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md (hunk 1 above)
Comment (matsduf) [reply docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md]: Corrected. Please re-review.

**Review by pnax — APPROVED** [review]

**Pull request merged**